storage/series/buffer.go

```diff
@@ -77,20 +78,20 @@ func newDatabaseBuffer(drainFn databaseBufferDrainFn, opts Options) databaseBuff
 }
 
 func (b *dbBuffer) Reset() {
-	b.forEachBucketAsc(b.nowFn(), func(bucket *dbBufferBucket, start time.Time) {
+	b.pastMostBucketIdx = b.forEachBucketAscToDrainOrReset(b.nowFn(), func(bucket *dbBufferBucket, start time.Time) {
```

Comment:
I agree with the lack of overloading functions to make it two functions, but can we make it more readable and clear by at least passing it as a typed enum? i.e. the 

```go
type computeReadWriteBucketIdx int

const (
  readBucketIdx computeReadWriteBucketIdx = iota
  computeBucketIdx
  computeAndWriteBucketIdx
)
```

Also can we call it `computedForEachBucketAsc` so it's more generic?